File: server/lab3/src/main/java/com/marcu/mobile/server/lab3/model/MusicalInstrument.java
package com.marcu.mobile.server.lab3.model;

import lombok.*;

import javax.persistence.*;

@Table(name = "musical_instruments")
@Entity
@Data
public class MusicalInstrument {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column
    private String name;

    @Column
    private String category;

    @Column
    private String description;

    @Column(name = "quantity_on_stock")
    private int quantityOnStock;

    @Column
    private double price;

    @Column
    private String currency;
}

File: native/lab1/app/src/main/java/com/example/lab1/MusicalInstrument.kt
/*
 * Copyright (C) 2017 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.lab1

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "musical_instruments")
data class MusicalInstrument(
    @PrimaryKey @ColumnInfo(name = "id") var id: Long,
    @ColumnInfo(name = "name") var name: String,
    @ColumnInfo(name = "category") var category: String,
    @ColumnInfo(name = "description") var description: String,
    @ColumnInfo(name = "quantity_on_stock") var quantityOnStock: Int,
    @ColumnInfo(name = "price") var price: Float,
    @ColumnInfo(name = "currency") var currency: String,
    @ColumnInfo(name = "synchronisedWithServer") var synchronisedWithServer: Boolean = true
)
